src/app-layout/index.tsx

```diff
@@ -266,19 +265,33 @@ const OldAppLayout = React.forwardRef(
       }
     );
 
-    const defaultDrawerSize = selectedDrawer?.size || toolsWidth;
+    const getSizes = () => {
+      const sizes: { [id: string]: number } = {};
+      if (!drawers) {
+        return {};
+      }
 
-    const [drawerSize = defaultDrawerSize, setDrawerSize] = useControllable(
-      selectedDrawer?.size,
+      for (const item of drawers.items) {
+        if (item.resizable && item.size) {
+          sizes[item.id] = item.size;
+        }
+      }
+      return sizes;
+    };
+
+    const [drawerSizes = {}, setDrawerSizes] = useControllable(
+      Object.keys(getSizes()).length > 0 ? getSizes() : undefined,
       drawers?.onResize,
-      defaultDrawerSize,
+      {},
       {
         componentName: 'AppLayout',
-        controlledProp: 'drawerSize',
-        changeHandler: 'onDrawerSize',
+        controlledProp: 'size',
+        changeHandler: 'onResize',
```

Comment:
I'd recommend to extend this text to reflect that the properties are inside another property. The strings here are only used for a console.log call, i.e. for human consumption. They do not need to match any specific value from Javascript's perspective, they just need to be useful for customers to understand where to find the properties. They are only used in these two messages:

https://github.com/cloudscape-design/components/blob/6f85b644a2990f0b0fd0395ce384f9a8e872158b/src/internal/hooks/use-controllable/index.ts#L65-L68

https://github.com/cloudscape-design/components/blob/6f85b644a2990f0b0fd0395ce384f9a8e872158b/src/internal/hooks/use-controllable/index.ts#L78-L84

For example something like 
> You provided a `drawers.items[n].size` prop without an `drawers.onResize` handler. This will render a non-interactive component.

by setting
```js
controlledProp: 'drawers.items[n].size',
changeHandler: 'drawers.onResize',
```